enumerables.rb

```diff
@@ -0,0 +1,162 @@
+
+      case arg[0]
+      when Class
+        condition = false if i.is_a?(arg[0]) == false
+      when Regexp
+        condition = false unless i&.to_s&.match?(arg[0])
+      when Integer || String
+        condition = false if arg[0] != i
+      when Proc
+        condition = false if arg[0].call(i) == false
+      end
+      result = yield(i) if block_given?
+      condition = result if block_given?
+      break if condition == false
+    end
+    condition
+  end
+
+  def my_any?(*arg)
+    arr = self.class == Range ? Array(self) : self
+
+    condition = false
+    return false if empty?
+
+    arr.my_each_with_index do |i, _j|
+      return true if i && !block_given? && !arg[0]
+
+      case arg[0]
+      when Class
+        condition = true if i.is_a?(arg[0])
+      when Regexp
+        condition = true if i&.to_s&.match?(arg[0])
+      when Integer || String
+        condition = true if arg[0] == i
+      end
+      result = yield(i) if block_given?
+      condition = result if block_given?
+      break if condition == true
+    end
+    condition
+  end
+
+  def my_none?(*arg)
+    arr = self.class == Range ? Array(self) : self
+    condition = true
+    return false if empty?
+
+    arr.my_each_with_index do |i, j|
+      return false if i && arr[j + 1] && arr[j] != arr[j + 1] && !arg[0] && !block_given?
+
+      case arg[0]
+      when Class
+        condition = false if i.is_a?(arg[0])
+      when Regexp
+        condition = true if i&.to_s&.match?(arg[0])
+      when Integer || String
+        condition = false if arg[0] == i
+      end
+      result = yield(i) if block_given?
+      condition = false if result && block_given?
+    end
+    condition
+  end
+
+  def my_count(arg = nil)
+    return length if !arg && !block_given?
+
+    counter = []
+    my_each do |i|
+      counter << i if (i == arg) && !block_given?
+      my_block = yield(i) if block_given?
+      counter << i if block_given? && my_block
+    end
+    counter.length
+  end
+
+  def my_map(prc = nil)
+    results = []
+    return to_enum(:my_map) unless prc || block_given?
+
+    my_each do |i|
+      my_block_map = yield(i) if block_given?
+
+      results << (prc ? prc.call(i) : my_block_map)
+    end
+    results
+  end
+
+  def my_inject(*param)
+    arr = self.class == Range ? Array(self) : self
+    return nil if self.class != Range && empty?
+
+    sym = nil
+    val = nil
+    sym = param[0] unless param.length == 2
+    if param.length == 2
+      sym = param[1]
+      val = param[0]
+    elsif block_given? && param[0]
+      arr << param[0]
+    end
+    temp = sym && sym.to_s == '*' ? 1 : 0
+
+    arr.my_each_with_index do |i, j|
+      temp = temp.send(sym, i) unless block_given?
+      if block_given? && arr[j + 1]
+        my_yield = yield(arr[j], arr[j + 1])
+        arr[j + 1] = my_yield
+        temp = my_yield
+      end
+    end
+    val ? temp.send(sym, val) : temp
+  end
+end
```

Comment:
### Mandatory
**`my_inject`** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)) 
- My inject can accept a Proc. and based on that it can perform operation. For an example->
```ruby
operation = proc { |sum, n| sum + n }
[1, 2 ,3].inject(&operation)  #should return 6
```